src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -126,7 +126,12 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
 
             if (autoDetectRegion)
             {
-                return await _regionDiscoveryProvider.GetMetadataAsync(new Uri(authority), requestContext).ConfigureAwait(false);
+                var regionalMetadata =  await _regionDiscoveryProvider.GetMetadataAsync(new Uri(authority), requestContext).ConfigureAwait(false);
+
+                //If the auto detected region is null, this indicates that the FallbackToGlobal parameter was set and the GetMetadataAsync failed to get the region.
```

Comment:
>/If the auto detected region is null, this indicates that the FallbackToGlobal parameter was set an [](start = 17, length = 99)

really? that sounds confusing. Are we keeping the right logic in the right place? when we need long explanations like this, naming and code location is often not the best choice.